discovery-provider/src/solana/anchor_program_indexer.py

```diff
@@ -133,4 +146,45 @@ def process_index_task(self):
     # parsed_transactions will contain an array of txs w/instructions
     # each containing relevant metadata in container
     async def fetch_ipfs_metadata(self, parsed_transactions):
-        return super().fetch_ipfs_metadata(parsed_transactions)
+
+        cid_to_user_id: Dict[str, int] = {}
+        cids_txhash_set: Set[Tuple(str, Any)] = set()
+        cid_type: Dict[str, str] = {}  # cid -> entity type track / user
```

Comment:
NOTE: `_fetch_metadata_from_gateway_endpoints` only takes in `track` and `user` but we will need `playlist` in the near future. 
No need to change now, but something to think about

Reply:
adding TODO there